libs/ballot-interpreter/src/hmpb-rust/timing_marks.rs

```diff
@@ -443,48 +459,130 @@ pub fn find_partial_timing_marks_from_candidate_rects(
     rects: &[Rect],
     debug: &ImageDebugWriter,
 ) -> Option<Partial> {
+    fn average<T: IntoIterator<Item = PixelPosition>>(values: T) -> PixelPosition {
+        let mut sum = 0;
+        let mut count = 0;
+        for value in values {
+            sum += value;
+            count += 1;
+        }
+        if count == 0 {
+            0
+        } else {
+            sum / count
+        }
+    }
+
     let half_height = (geometry.canvas_size.height / 2) as PixelPosition;
     let half_width = (geometry.canvas_size.width / 2) as PixelPosition;
     let top_half_rects = rects
         .iter()
         .filter(|r| r.top() < half_height)
         .copied()
-        .collect::<Vec<Rect>>();
+        .collect::<Vec<_>>();
```

Comment:
This doesn't change the functionality. It just tells Rust "you figure out the type where `_` appears".

Reply:
Yes, but the idiomatic approach in Rust is usually to let the compiler infer types when it can. In this case the inferred `_` has to be `Rect` and is specified elsewhere, so there’s no need to duplicate it.